core/src/main/java/org/apache/iceberg/DefaultModifyFiles.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterables;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.expressions.Evaluator;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Projections;
+
+public class DefaultModifyFiles extends MergingSnapshotProducer<ModifyFiles> implements ModifyFiles {
```

Comment:
In other cases, we use `Base` for a prefix on the built-in implementation. I know Iceberg is not very consistent for the other operations, but it would be good to standardize on one name prefix for this.

Reply:
It would be really great to have a consistent naming of basic implementations. Right now, we have synonyms in many places (e.g. `RewriteFiles` -> `ReplaceFiles`, `RewriteManifests` -> `ReplaceManifests`, `OverwriteFiles` -> `OvewriteData`). That might be confusing if you don't know what is an interface and what is implementation. I think it would be better to have `BaseRewriteFiles`, `BaseOverwriteFiles`, etc.

Reply:
I think it's a good idea to rename these to be consistent.